mne/io/neuralynx/neuralynx.py

```diff
@@ -98,31 +139,133 @@ def __init__(
         # the sample sizes of all segments
         n_segments = nlx_reader.header["nb_segment"][0]
         block_id = 0  # assumes there's only one block of recording
-        n_total_samples = sum(
-            nlx_reader.get_signal_size(block_id, segment)
-            for segment in range(n_segments)
+
+        # get segment start/stop times
+        start_times = np.array(
+            [nlx_reader.segment_t_start(block_id, i) for i in range(n_segments)]
+        )
+        stop_times = np.array(
+            [nlx_reader.segment_t_stop(block_id, i) for i in range(n_segments)]
         )
 
-        # construct an array of shape (n_total_samples,) indicating
-        # segment membership for each sample
-        sample2segment = np.concatenate(
-            [
-                np.full(shape=(nlx_reader.get_signal_size(block_id, i),), fill_value=i)
-                for i in range(n_segments)
+        # find discontinuous boundaries (of length n-1)
+        next_start_times = start_times[1::]
+        previous_stop_times = stop_times[:-1]
+        seg_diffs = next_start_times - previous_stop_times
+
+        # mark as discontinuous any two segments that have
+        # start/stop delta larger than sampling period (1/sampling_rate)
+        logger.info("Checking for temporal discontinuities in Neo data segments.")
+        delta = 1 / info["sfreq"]
+        gaps = seg_diffs > delta
+        has_gaps = gaps.any()
+
+        seg_gap_dict = {}
+
+        if has_gaps:
```

Comment:
Rather than three separate conditional `if has_gaps`, I wonder if we can get away with a single code path. In other words, make things a *tiny* bit less efficient by always doing the `gap_starts` etc. below and never have a conditional an `has_gaps`, and always constructing `Annotations`. And in the case where there are no gaps, these should become no-ops anyway, i.e., the `Annotations` will be empty (which is fine).

Usually code can be constructed this way and 1) it's a nice sanity check that it's coded properly, and 2) fewer branches mean less stuff to understand / think about when debugging. But if you try to implement things this way and it becomes a pain then I can live with the current branching!

Reply:
Ok, got it! Will give it a shot! Thanks.

Reply:
It seems branches were indeed unnecessary. Can make tests passing after removing the conditionals and adjusting accordingly in https://github.com/mne-tools/mne-python/pull/12279/commits/37197c7b18546bb77af7e075d691594d8ee1e854.